### PR TITLE
auto sync tf config with checkly backend

### LINKED ISSUE #11: Manually deleting check gives API error
Migrated from https://github.com/bitfield/terraform-provider-checkly/issues/9

Steps to reproduce:

Create check via terraform
Delete check via UI
run terraform apply or terraform plan

Result:

checkly_check.example-check: Refreshing state... [id=36152cfc-f65c-42a9-b922-7eadd23c5840]

Error: API error: unexpected response status 404: "{\"statusCode\":404,\"error\":\"Not Found\"}"

Confirmed this still happens on v0.6.0

### PR DESCRIPTION
**Pull request by kjda**
## Affected Components
* [X] Resources
* [ ] Tests
* [ ] Docs
* [] Other

## Style
* [x] Terraform code is formatted with `terraform fmt`
* [x] Go code is formatted with `go fmt`

fixes #11

### CODE REVIEW

File: resource_check.go [hunk 1]
```diff
@@ -336,19 +337,42 @@ func resourceCheckCreate(d *schema.ResourceData, client interface{}) error {
...
+	return readCheck(d, client, true)
+}
+
+func readCheck(d *schema.ResourceData, client interface{}, shouldSyncRemote bool) error {
 	check, err := client.(*checkly.Client).Get(d.Id())
 	if err != nil {
-		return fmt.Errorf("API error 2: could not read check %s, Error: %w", d.Id(), err)
+		if shouldSyncRemote && strings.Contains(err.Error(), "404") {
+			// the resource was deleted remotely, try to recreate it
+			{
```
Comment (umutuzgur) [inline resource_check.go:359]: What do we have the extra brackets? Do you want to turn this into another func for readability?

Comment (kjda) [reply resource_check.go:359]: it's just for readability

Comment (umutuzgur) [reply resource_check.go:359]: Could you turn this into a function? I find that easier to read personally

Comment (kjda) [reply resource_check.go:359]: I don't think it makes sense to turn this into a function, putting comment and belonging code in extra brackets is a style for easily indicating comment scope

Comment (umutuzgur) [reply resource_check.go:359]: We can go with this then but, in general, I'm not a big fan of code blocks with brackets to comment and abstract code. I find methods and functions more suitable for this

Comment (kjda) [reply resource_check.go:359]: I agree, but depending on the situation, this code block basically does a simple `if then set` & it seem to belong to the local function scope logically, maybe the brackets here were too much, normally they make sense when the comment cover more than one statement, but here it's just about a single if statement

**Pull request merged**